File: eShopLegacyWebFormsSolution/eShopLegacyWebForms-SpringBoot/src/main/java/com/eshop/webforms/dto/PaginatedItemsViewModel.java
package com.eshop.webforms.dto;

import java.util.List;

public class PaginatedItemsViewModel<T> {

    private final int actualPage;
    private final int itemsPerPage;
    private final long totalItems;
    private final int totalPages;
    private final List<T> data;

    public PaginatedItemsViewModel(int pageIndex, int pageSize, long count, List<T> data) {
        this.actualPage = pageIndex;
        this.itemsPerPage = pageSize;
        this.totalItems = count;
        this.totalPages = (int) Math.ceil((double) count / pageSize);
        this.data = data;
    }

    public int getActualPage() {
        return actualPage;
    }

    public int getItemsPerPage() {
        return itemsPerPage;
    }

    public long getTotalItems() {
        return totalItems;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public List<T> getData() {
        return data;
    }
}

File: modernized/src/Catalog.Mvc/Models/PaginatedItemsViewModel.cs
namespace Catalog.Mvc.Models;

/// <summary>
/// Server-side paging view model ported from the legacy MVC
/// <c>eShopLegacyMVC.ViewModel.PaginatedItemsViewModel&lt;TEntity&gt;</c>.
/// Preserves the same fields and the <c>TotalPages = ceil(count / pageSize)</c>
/// calculation so paging behavior matches the behavioral baseline.
/// </summary>
public class PaginatedItemsViewModel<TEntity>
    where TEntity : class
{
    public PaginatedItemsViewModel(int pageIndex, int pageSize, long count, IEnumerable<TEntity> data)
    {
        ActualPage = pageIndex;
        ItemsPerPage = pageSize;
        TotalItems = count;
        TotalPages = (int)Math.Ceiling((decimal)count / pageSize);
        Data = data;
    }

    public int ActualPage { get; }

    public int ItemsPerPage { get; }

    public long TotalItems { get; }

    public int TotalPages { get; }

    public IEnumerable<TEntity> Data { get; }
}
